--- webserver.orig.c	2023-04-02 17:31:11.477262000 -0700
+++ webserver.c	2023-04-02 19:19:45.361646000 -0700
@@ -34,6 +34,8 @@
 #include <pthread.h>
 
 #define _XOPEN_SOURCE
+#include <time.h>
+char *strptime(const char *s, const char *format, struct tm *tm);
 
 typedef struct {
 	char *method;
@@ -82,6 +84,9 @@
 	strcat(searchstr, ": ");
 
 	if (hdrptr = strstr(req->headers, searchstr)) { 
+		if (strlen(hdrptr) > 1000)
+			return NULL;
+		
 		hdrptr += strlen(searchstr); 
 		if (hdrend = strstr(hdrptr, "\r\n")) { 
 			char hdrval[1024]; // temporary return value
